### PR TITLE
Feat/failed reasons be

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Add column failed_reason to order entity.
- Create API endpoint for setting failed order reason.

photo_2024-12-23_21-39-35

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-143)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

- [x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [ ] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/common/database/entities/order.entity.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export class Order {
   @Column({ nullable: false, default: 'files/tickets/image 30.png' })
   ticket_photo: string;
 
+  @Column({ type: 'text', nullable: true })
```
Comment (Parle648) [inline src/common/database/entities/order.entity.ts:38]: where is the updated entity ( screenshot ) in PR description?

File: src/modules/orders/orders.service.ts [hunk 2]
```diff
@@ -293,4 +293,23 @@ export class OrdersService {
       throw new InternalServerErrorException('Internal Server Error');
     }
   }
+
+  async setFailedReason(orderId: number, reason: string): Promise<Order> {
+    try {
+      const order = await this.orderRepository.findOne({ where: { id: orderId } });
+
+      if (!order) {
```
Comment (Parle648) [inline src/modules/orders/orders.service.ts]: There is findOneOrFail or findOneByOrFail, anyway I remember that there is such method that allows us handle this verification automatically

File: src/common/database/entities/order.entity.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/common/database/entities/order.entity.ts:38]: screen added

File: src/modules/orders/orders.service.ts (hunk 2 above)
Comment (oyakovytskyi) [reply src/modules/orders/orders.service.ts]: replace to findOneOrFail

**Review by Parle648 — APPROVED** [review]

**Review by AlexeyBezrebriy — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**